src/databricksApi/dbfs/DatabricksFSTreeItem.ts

```diff
@@ -128,7 +143,35 @@ export class DatabricksFSTreeItem extends vscode.TreeItem implements iDatabricks
 		{
 			throw new Error("Only a single file can be downloaded!");
 		}
-		
+	}
+
+	async upload(): Promise<void> {
+		if(!this.is_dir)
+		{
+			try {
```

Comment:
ideally, if the local file does not exist, the upload-button is not shown at all
this is already controlled by the VIEWSTATE (or contextValue property) and it should not be necessary to check it here again?

Reply:
You are right - removed the lines.